src/network/pages/SponsorshipsPage.tsx

```diff
@@ -101,12 +118,23 @@ export const SponsorshipsPage = () => {
                 rightSideContent={
                     <Button
                         onClick={async () => {
-                            try {
-                                await createSponsorshipModal.pop()
-                            } catch (e) {
-                                // Ignore for now.
+                            if (balanceData) {
+                                try {
+                                    await createSponsorshipModal.pop({
+                                        balance: balanceData.balance,
+                                        tokenSymbol: balanceData.tokenSymbol,
+                                        tokenDecimals: balanceData.tokenDecimals,
+                                        onSubmit: handleSponsorshipCreation,
+                                        onResolve: () => {
+                                            sponsorshipsQuery.refetch()
+                                        },
+                                    })
```

Comment:
Let's tighty it up a little.

```tsx
try {
    await createSponsorshipModal.pop(/* … */)

    // Everything here is `onResolve`.

    sponsorshipsQuery.refetch()
} catch (e) {
    // Everything here is `onReject`.

    console.warn('¯\_(ツ)_/¯')
}
```

So

```suggestion
                                    await createSponsorshipModal.pop({
                                        balance: balanceData.balance,
                                        tokenSymbol: balanceData.tokenSymbol,
                                        tokenDecimals: balanceData.tokenDecimals,
                                        onSubmit: handleSponsorshipCreation,
                                    })

                                    sponsorshipsQuery.refetch()
```

Reply:
Is `sponsorshipsQuery.refetch()` async, too?